src/components/header/header.spec.js

```diff
@@ -0,0 +1,116 @@
+import { renderComponent, setTestPage } from '../../tests/helpers/rendering';
+import { EXAMPLE_HEADER_SEARCH_AND_MENU_LINKS } from './_test-examples';
+import { getNodeAttributes } from '../../tests/helpers/puppeteer';
+
+describe('script: header', () => {
+    beforeEach(async () => {
+        await setTestPage('/test', renderComponent('header', { ...EXAMPLE_HEADER_SEARCH_AND_MENU_LINKS, variants: 'basic' }));
+    });
+
+    describe('WHEN: the search button is clicked', () => {
+        beforeEach(async () => {
+            await page.click('.ons-js-toggle-header-search');
+        });
+
+        it('THEN: the search input form is displayed', async () => {
+            await page.waitForSelector('.ons-header-nav-search');
+            const isSearchNavVisible = await page.$eval('.ons-header-nav-search', (el) => !el.classList.contains('ons-u-d-no'));
+            expect(isSearchNavVisible).toBe(true);
+        });
+
+        it('THEN: the button switches to the close icon', async () => {
+            const attributes = await getNodeAttributes(page, '.ons-js-toggle-header-search');
+            expect(attributes['aria-expanded']).toBe('true');
+
+            const hasCloseIcon = await page.$eval('.ons-js-toggle-header-search svg', (el) => el.classList.contains('ons-icon--close'));
+            expect(hasCloseIcon).toBe(true);
+        });
+
+        it('THEN: the search field has full width', async () => {
+            await page.waitForSelector('.ons-header-nav-search__input .ons-input');
+            const isFullWidth = await page.$eval('.ons-header-nav-search__input .ons-input', (el) =>
+                el.classList.contains('ons-input--w-full'),
+            );
+            expect(isFullWidth).toBe(true);
+        });
+    });
+
+    describe('WHEN: the search button is NOT clicked', () => {
```

Comment:
We should also test that aria-expanded is set correctly. And again when the button is clicked